### PR TITLE
Add onDisposed to Maybe, Single and Completable

### PR DESCRIPTION
**Pull request by MortyMerr**
Based on https://github.com/ReactiveX/RxSwift/pull/1681 which is looks like forgotten

### CODE REVIEW

**Comment by kzaher** [comment]
I think we should add this, but we should either create new convenience with non optional `onDisposed` or wait for 6.0 because this breaks the ABI.

**Comment by MortyMerr** [comment]
@kzaher thank you for your attention :)
Why `onDisposed` should be non-optional?

**Comment by kzaher** [comment]
Because we already have `subscribe(onCompleted: (() -> Void)? = nil, onError: ((Swift.Error) -> Void)? = nil)`.

**Comment by MortyMerr** [comment]
Just to clarify, you mean, we have to create an additional `subscribe` method which will take `onDisposed` as non-optional parameter to prevent breaking the ABI?

File: RxSwift/Traits/Completable.swift [hunk 1]
```diff
@@ -76,27 +76,46 @@ extension PrimitiveSequenceType where Trait == CompletableTrait, Element == Swif
...
         #else
                 let callStack = [String]()
         #endif
 
-        return self.primitiveSequence.subscribe { event in
+        let disposable: Disposable
+        if let disposed = onDisposed {
+            disposable = Disposables.create(with: disposed)
+        }
+        else {
```
Comment (freak4pc) [inline RxSwift/Traits/Completable.swift]: Change to `} else {` please (e.g. move up a line)
```suggestion
        else {
```

File: RxSwift/Traits/Completable.swift [hunk 2]
```diff
@@ -76,27 +76,46 @@ extension PrimitiveSequenceType where Trait == CompletableTrait, Element == Swif
...
+                          onDisposed: (() -> Void)? = nil) -> Disposable {
         #if DEBUG
                 let callStack = Hooks.recordCallStackOnError ? Thread.callStackSymbols : []
         #else
                 let callStack = [String]()
         #endif
 
-        return self.primitiveSequence.subscribe { event in
+        let disposable: Disposable
+        if let disposed = onDisposed {
```
Comment (freak4pc) [inline RxSwift/Traits/Completable.swift]:
```suggestion
        if let onDisposed = onDisposed {
```

**Comment by t4rj4n** [comment]
Hello,
could it be, that this feature never managed to get into the master branch?
I cannot find it here:
https://github.com/ReactiveX/RxSwift/blob/master/RxSwift/Traits/Completable.swift
And if im reading it here correctly, the pull request errored in to CI:
https://travis-ci.org/github/ReactiveX/RxSwift/builds/632257143

Greetings from t4rj4n

**Comment by freak4pc** [comment]
I think this commit is part of the RxSwift 6 branch. I might cherry-pick some of the commits into the master branch since they can work absolutely fine on RxSwift 6.

**Pull request merged**